extensions/terminal-suggest/src/terminalSuggestMain.ts

```diff
@@ -289,14 +289,17 @@ export async function collectCompletionItemResult(command: Command, parsedArgume
 		if (!specArgs) {
 			return { filesRequested, foldersRequested };
 		}
-
+		const excludeSet = kind === vscode.TerminalCompletionItemKind.Flag ? parsedArguments?.passedOptions.map(option => option.name).flat() : undefined;
 		if (Array.isArray(specArgs)) {
 			for (const item of specArgs) {
 				const suggestionLabels = getLabel(item);
-				if (!suggestionLabels) {
+				if (!suggestionLabels?.length) {
 					continue;
 				}
 				for (const label of suggestionLabels) {
+					if (excludeSet?.includes(label)) {
```

Comment:
Could be simplified a little by added above:

```ts
const passedOptions = parsedArguments?.passedOptions
```

Then here doing:

```ts
if (passedOptions?.find(e => e.name === label)) {
```

Reply:
it's used below too

Reply:
https://github.com/microsoft/vscode/blob/8c4c086235074300ee01ce7cf2a27cce4671eeed/extensions/terminal-suggest/src/terminalSuggestMain.ts#L317

Reply:
renamed the variable